__tests__/utils/storage.test.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { MemoryStore, Storage } from '../../src/storage';
+import { scanPolicyStartAddresses, checkScanningPolicy } from '../../src/utils/storage';
+
+describe('scanning policy methods', () => {
+  it('start addresses', async () => {
+    const store = new MemoryStore();
+    const storage = new Storage(store);
+    const gapLimit = 27;
+    jest.spyOn(storage, 'getGapLimit').mockReturnValue(Promise.resolve(gapLimit));
+    jest.spyOn(storage, 'getScanningPolicy').mockReturnValue(Promise.resolve('gap-limit'));
+    await expect(scanPolicyStartAddresses(storage)).resolves.toEqual({
+      nextIndex: 0,
+      count: gapLimit,
+    });
+  });
+
+  it('check address scanning policy', async () => {
+    const store = new MemoryStore();
+    const storage = new Storage(store);
+    const gapLimit = 27;
+    jest.spyOn(storage, 'getGapLimit').mockReturnValue(Promise.resolve(gapLimit));
+    const policyMock = jest.spyOn(storage, 'getScanningPolicy');
+
+    policyMock.mockReturnValue(Promise.resolve('gap-limit'));
+    await expect(checkScanningPolicy(storage)).resolves.toEqual({
+      nextIndex: 1,
+      count: 19,
+    });
```

Comment:
question: If this wallet is pristine and the gapLimit is `27`, why is the `count` only `19`?
Isn't it supposed to be `26`? ( [source code](https://github.com/HathorNetwork/hathor-wallet-lib/blob/7e06f89d2a62abbe01656d0dcb1313b28008f54f/src/utils/storage.ts#L304) )
```js
lastUsedAddressIndex + gapLimit - lastLoadedAddressIndex,
// -1 + 27 - 0
```

Reply:
Nice catch, the test was incorrectly mocking the value.